src/services/smart-fields-values-injector.js

```diff
@@ -17,6 +17,25 @@ function SmartFieldsValuesInjector(record, modelName, fieldsPerModel) {
       }
     }
 
+    // NOTICE: Update the record with smart field value (contains side effects).
```

Comment:
Good catch, thank you!  I created this function to avoid duplication, but it resulted in something weird!

So, based on your comment, I _removed_ this procedural function that performed a sequence of actions.  It result in a more concise code (`+38 / -45`), easier to read _in my opinion_ (considering the Smart Field value as a promise, I can now read the whole code vertically from top to bottom: it actually work well with my mind ; I hope it will work for you too!).